tests/root/arch/unit_testing/CMakeLists.txt

```diff
@@ -0,0 +1,5 @@
+# SPDX-License-Identifier: Apache-2.0
+#
+# Copyright (c) 2022 Nordic Semiconductor
+
+# Intentionally left empty
```

Comment:
Why are all these under `tests/` instead of at the top level `arch/unit_testing/`? I wouldn't have found it if I wasn't reviewing this PR

Reply:
because if they are placed under toplevel `arch/unit_testing` then that arch is always available, and I really don't like that.

The `unit_testing` board and arch are not really board and arch in the normal sense, instead they are solving a very special testing use-case.

But i'm open to another location than `tests/root/<board|arch>` cause i'm not particularly fond of that location either.
Just lack a better candidate, as `unit_testing` should not pollute `<ZEPHYR_BASE>/boards` and `<ZEPHYR_BASE>/arch`

Reply:
Can we put it under `subsys/testsuite/arch/`?

Reply:
Not a bad location.

So then we would have:
`subsys/testsuite/arch/`
`subsys/testsuite/boards/`

for what is related to unittesting.

Agreed ?

Reply:
Yup LGTM

Reply:
done.

Reply:
Thanks!